eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/ExponentialRetry.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import java.time.Duration;
+
+public final class ExponentialRetry extends Retry {
+
+    public static final Duration TIMER_TOLERANCE = Duration.ofSeconds(1);
+    private final Duration minBackoff;
+    private final Duration maxBackoff;
+    private final double retryFactor;
+
+    /**
+     * @param minBackoff The minimum time period permissible for backing off between retries.
+     * @param maxBackoff The maximum time period permissible for backing off between retries.
+     * @param maxRetryCount The maximum number of retries allowed.
+     */
+    public ExponentialRetry(Duration minBackoff, Duration maxBackoff, int maxRetryCount) {
+        super(maxRetryCount);
+        this.minBackoff = minBackoff;
+        this.maxBackoff = maxBackoff;
+
+        this.retryFactor = computeRetryFactor();
+    }
+
+    @Override
+    protected Duration calculateNextRetryInterval(final Exception lastException,
+                                              final Duration remainingTime,
+                                              final int baseWaitSeconds,
+                                              final int retryCount) {
+        if ((!Retry.isRetriableException(lastException)) || retryCount >= super.maxRetryCount()) {
+            return null;
+        }
+        final double nextRetryInterval = Math.pow(retryFactor, (double) retryCount);
+
```

Comment:
nit: random blank line. I try to keep my variable declarations together, leave a blank line and then start invoking methods, etc on them.